frontend/src/scenes/billing/billingUsageLogic.ts

```diff
@@ -0,0 +1,274 @@
+import { actions, afterMount, connect, kea, key, listeners, path, props, reducers, selectors } from 'kea'
+import { loaders } from 'kea-loaders'
+import { subscriptions } from 'kea-subscriptions'
+import api from 'lib/api'
+import { dayjs } from 'lib/dayjs'
+import { dateMapping, toParams } from 'lib/utils'
+import { organizationLogic } from 'scenes/organizationLogic'
+
+import type { OrganizationType } from '~/types'
+import { BillingType, DateMappingOption } from '~/types'
+
+import { billingLogic } from './billingLogic'
+import type { billingUsageLogicType } from './billingUsageLogicType'
+import { ALL_USAGE_TYPES } from './constants'
+
+export interface BillingUsageResponse {
+    status: 'ok'
+    type: 'timeseries'
+    customer_id: string
+    results: Array<{
+        id: number
+        label: string
+        data: number[]
+        dates: string[]
+        breakdown_type: 'type' | 'team' | 'multiple' | null
+        breakdown_value: string | string[] | null
+    }>
+    next?: string
+}
+
+export interface BillingUsageFilters {
+    usage_types?: string[]
+    team_ids?: number[]
+    breakdowns?: ('type' | 'team')[]
+    interval?: 'day' | 'week' | 'month'
+}
+
+export const DEFAULT_BILLING_USAGE_FILTERS: BillingUsageFilters = {
+    breakdowns: ['type'],
+    usage_types: [],
+    team_ids: [],
+    interval: 'day',
+}
+
+export interface BillingUsageLogicProps {
+    dashboardItemId?: string
+}
+
+export const billingUsageLogic = kea<billingUsageLogicType>([
+    path(['scenes', 'billing', 'billingUsageLogic']),
+    props({} as BillingUsageLogicProps),
+    key(({ dashboardItemId }) => dashboardItemId || 'global'),
+    connect({
+        values: [organizationLogic, ['currentOrganization'], billingLogic, ['billing']],
+    }),
+    subscriptions((logic: billingUsageLogicType) => ({
+        currentOrganization: (org: OrganizationType | null, prevOrg: OrganizationType | null) => {
```

Comment:
I'm surprised this is needed. I would have thought the logic would remount when changing organizations.

Reply:
You're right, not needed anymore